zerver/lib/retention.py

```diff
@@ -225,24 +223,16 @@ def move_expired_personal_and_huddle_messages_to_archive(
     check_date = timezone_now() - timedelta(days=message_retention_days)
 
     # This function will archive appropriate messages and their related objects.
-    internal_realm = get_realm(settings.SYSTEM_BOT_REALM)
-    cross_realm_bot_ids = [
-        get_user_including_cross_realm(email, internal_realm).id
-        for email in settings.CROSS_REALM_BOT_EMAILS
-    ]
     recipient_types = (Recipient.PERSONAL, Recipient.HUDDLE)
 
-    # Archive expired personal and huddle Messages in the realm, except cross-realm messages.
-    # The condition zerver_userprofile.realm_id = {realm_id} assures the row won't be
-    # a message sent by a cross-realm bot, because cross-realm bots have their own separate realm.
+    # Archive expired personal and huddle Messages in the realm, including cross-realm messages.
     query = SQL(
         """
     INSERT INTO zerver_archivedmessage ({dst_fields}, archive_transaction_id)
         SELECT {src_fields}, {archive_transaction_id}
         FROM zerver_message
         INNER JOIN zerver_recipient ON zerver_recipient.id = zerver_message.recipient_id
-        INNER JOIN zerver_userprofile ON zerver_userprofile.id = zerver_message.sender_id
-        WHERE zerver_userprofile.realm_id = {realm_id}
+        WHERE zerver_message.realm_id = {realm_id}
```

Comment:
This query is probably not fantastic from a query plan perspective; but I'm actually not sure how to build an effective index for this, since we'd want the index to be on something like `(realm, recipient__type, date_sent)`

It can't really be worse to get rid of the UserProfile join in any case, though.

Reply:
That should be straightfoward if we go through with the `message.type` migration though, right?